src/libsystemd/sd-network/network-util.h

```diff
@@ -44,6 +44,15 @@ typedef enum LinkAddressState {
         _LINK_ADDRESS_STATE_INVALID = -EINVAL,
 } LinkAddressState;
 
+typedef enum LinkOnlineState {
+        LINK_ONLINE_STATE_UNKNOWN,
+        LINK_ONLINE_STATE_OFFLINE,
+        LINK_ONLINE_STATE_PARTIALLY_ONLINE,
+        LINK_ONLINE_STATE_ONLINE,
+        _LINK_ONLINE_STATE_MAX,
+        _LINK_ONLINE_STATE_INVALID = -EINVAL,
```

Comment:
hmm, do we really want both LINK_ONLINE_STATE_UNKNOWN and _LINK_ONLINE_STATE_INVALID? The latter should suffice as marker for "dont know", no?

Reply:
the intent here was to make it a "first class citizen" among statuses - it seems a little weird to repurpose _LINK_ONLINE_STATE_INVALID for that. I'll try to change it in the next revision and we can see how it looks

Reply:
I removed the unknown state and use _LINK_ONLINE_STATE_INVALID now. Seems to work just fine.